examples/webgl_depth_peeling.html

```diff
@@ -0,0 +1,305 @@
+								// samples: 2,
+							} )
+						);
+
+					this.depth = depth;
+
+	}
+				setScreenSize( width, height, pixelRatio ) {
+
+					this.screenSize.set( width, height );
+					this.pixelRatio = pixelRatio;
+					const w = width * pixelRatio;
+					const h = height * pixelRatio;
+					this.globalUniforms.uReciprocalScreenSize.value.set(
+						1 / w,
+						1 / h
+					);
+
+					this.layers.forEach( ( rt ) => {
+
+						rt.setSize( w, h );
+						rt.depthTexture.dispose();
+						rt.depthTexture = new THREE.DepthTexture( w, h );
+
+					} );
+
+	}
+
+			}
+
+
+
+			const AppState = {
+				Enabled: true,
+				Depth: 3,
+			};
+			init();
+			async function init() {
+
+				const renderer = new THREE.WebGLRenderer();
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				document.body.appendChild( renderer.domElement );
+
+				const scene = new THREE.Scene();
+				const width = window.innerWidth, height = window.innerHeight;
+				const camera = new THREE.PerspectiveCamera( 75, width / height, 0.1, 1000 );
+				camera.position.z = 5;
+
+				renderer.shadowMap.enabled = true;
+
+				const dirLight = new THREE.DirectionalLight();
+				dirLight.position.set( 0, 3, 0 );
+				dirLight.castShadow = true;
+				scene.add( dirLight );
+				scene.add( new THREE.AmbientLight( undefined, 0.5 ) );
+
+				const sphere = new THREE.Mesh(
+					new THREE.SphereBufferGeometry(),
+					new THREE.MeshStandardMaterial()
+				);
+				sphere.translateY( 3 ).translateX( 1.5 );
+				sphere.castShadow = true;
+				scene.add( sphere );
+
+				const knot = new THREE.Mesh(
+					new THREE.TorusKnotBufferGeometry( undefined, undefined, 128, 32 ),
+					new THREE.MeshStandardMaterial( {
+						transparent: true,
+						opacity: 0.7,
+						side: THREE.DoubleSide,
+					} )
+				);
+				knot.receiveShadow = true;
+				scene.add( knot );
+
+				scene.add(
+					new THREE.Mesh(
+						new THREE.BoxBufferGeometry(),
+						new THREE.MeshBasicMaterial( { color: 0xf0a000 } )
+					)
+				);
+
+				const texture = await new THREE.TextureLoader().loadAsync(
+					'textures/sprite0.png'
+				);
+				const plane = new THREE.Mesh(
+					new THREE.PlaneBufferGeometry( 3, 3 ),
+					new THREE.MeshStandardMaterial( { map: texture, side: THREE.DoubleSide } )
+				);
+				plane.translateX( - 1.6 );
+				plane.translateY( 1.5 );
+				scene.add( plane );
+
+				const plane2 = new THREE.Mesh(
+					new THREE.PlaneBufferGeometry( 3, 3 ),
+					new THREE.MeshStandardMaterial( {
+						map: texture,
+						transparent: true,
+						side: THREE.DoubleSide,
+					} )
+				);
+				plane2
+					.translateX( - 1.2 )
+					.translateY( - 1.5 )
+					.translateZ( 0 )
+					.rotateY( - 2 * Math.PI * ( 1 / 10 ) );
+				scene.add( plane2 );
+
+				const dp = new DepthPeeling( { depth: AppState.Depth, width: window.innerWidth, height: window.innerHeight, pixelRatio: renderer.getPixelRatio() } );
+				dp.add( scene );
```

Comment:
TBH, the structure and usage of the `DepthPeeling` class is a bit unusual consider the existing APIs in `three.js`. It does not match the API of effects classes like `StereoEffect` and it also not fits in the post processing area.

How about structuring the example without a `DepthPeeling` class. That should be a better start since it would make it easier for users to integrate the logic in their app. In the long term, depth peeling is a technique that should be implemented inside the renderer (and maybe just enabled with a flag).

Reply:
Sure! I'll work on it

Reply:
@Mugen87 I dropped DepthPeeling class and refactored to be similar to webgl_rtt